### PR TITLE
Fix integration with completion, resolve static variable access on instances, docblock on interfaces

### PR DESCRIPTION
**Pull request by BladeMF**
1. For variables: resolve the parent `ScopedPropertyAccessExpression` only if the variable is the memberName.
2. When resolving `ScopedPropertyAccessExpression` see if `scopeResolutionQualifier` is a `Variable` and then resolve it to get the type, rather than taking the text and treating it as a type.
3. Ignore smoke_test.log
4. Don't search for properties in docblocks on interfaces

### CODE REVIEW

File: lib/Bridge/Phpactor/Docblock.php [hunk 1]
```diff
@@ -158,6 +159,9 @@ public function properties(ReflectionClassLike $declaringClass): ReflectionPrope
             if (!$propertyTag->propertyName()) {
                 continue;
             }
+            if ($declaringClass instanceof ReflectionInterface) {
+                continue;
+            }
```
Comment (dantleech) [inline lib/Bridge/Phpactor/Docblock.php:164]: :+1:

File: lib/Core/Inference/SymbolContextResolver.php [hunk 2]
```diff
@@ -597,7 +601,23 @@ private function resolveSubscriptExpression(
...
     {
-        $name = $node->scopeResolutionQualifier->getText();
+        $name = null;
+        if ($node->scopeResolutionQualifier instanceof ParserVariable) {
+            /** @var SymbolContext $context */
+            $context = $this->resolveVariable($frame, $node->scopeResolutionQualifier);
+            if (
+                $context->types() !== null &&
+                $context->types()->count() > 0 &&
+                $context->type()->className() !== null
```
Comment (dantleech) [inline lib/Core/Inference/SymbolContextResolver.php:611]: `SymbolContext->types()` cannot be `null`, neither can `type()` (it returns a Null object). It should suffice to just just if `className` is `!=null`

**Comment by dantleech** [comment]
nice :+1: we can probably remove some the NULL checks as indicated above.

**Comment by dantleech** [comment]
btw - not important but in the future you can use `git rebase origin/master` instead of `git merge origin/master` to avoid the `Merge` commits. It's better to use rebase on feature branches.

**Comment by dantleech** [comment]
It fixed the completion, thanks !

**Comment by BladeMF** [comment]
Fixes https://github.com/phpactor/worse-reflection/issues/86

**Pull request merged**